api/src/main/java/org/openmrs/validator/PersonValidator.java

```diff
@@ -107,7 +108,36 @@ public void validate(Object target, Errors errors) {
 		
 		ValidateUtil.validateFieldLengths(errors, Person.class, "gender", "personVoidReason");
 	}
-	
+
+	/**
+	 * Checks if the death date is in accordance with dead status.
+	 *
+	 * @param errors Stores information about errors encountered during validation
+	 * @param deathDate the deathdate to validate
+	 * @param dead is the dead status
+	 */
+	private void validateDeathDateDead(Errors errors, Date deathDate, boolean dead) {
+		if(deathDate == null || dead == true) {
```

Comment:
if the person is dead, we dont care if deathdate is set or not, if its set its fair, if not fair too.
if deathdate is null we dont care if person is dead or not.
but if deathdate is set the peron should surely be dead.